src/workerd/io/worker-entrypoint.c++

```diff
@@ -438,7 +438,129 @@ kj::Promise<void> WorkerEntrypoint::request(
 kj::Promise<void> WorkerEntrypoint::connect(kj::StringPtr host, const kj::HttpHeaders& headers,
     kj::AsyncIoStream& connection, ConnectResponse& response,
     kj::HttpConnectSettings settings) {
-  JSG_FAIL_REQUIRE(TypeError, "Incoming CONNECT on a worker not supported");
+  auto incomingRequest = kj::mv(KJ_REQUIRE_NONNULL(this->incomingRequest,
```

Comment:
Likely yes. For now I'd rather keep things separate. We can revisit and eliminate duplication in a future PR

Reply:
I really don't think we should introduce a whole bunch of duplicated code which will inevitably get out-of-sync long before anyone cleans it up.

Reply:
Before doing any deduplication I'd like the code review to focus on correctness. Is the code doing the right thing in the way it needs to be done. I'll reconcile the duplicated bits of code as a follow up.

Reply:
I can't review the correctness of this code as-is. I would have to manually diff it against the code it was copied from. I need you to factor out the shared stuff so that I can adequately review it.